### PR TITLE
types: improve webpack plugin types

### PR DESCRIPTION
**Pull request by lmiller1990**
Some TS tips.

### CODE REVIEW

File: npm/webpack-dev-server/src/plugin.ts [hunk 1]
```diff
@@ -28,11 +28,7 @@ export default class CypressCTOptionsPlugin implements Plugin {
     this.devserverEvents = options.devserverEvents
   }
 
-  private pluginFunc = (context: CypressCTWebpackContext, module) => {
-    if (!module.resource) {
```
Comment (lmiller1990) [inline npm/webpack-dev-server/src/plugin.ts:32]: why did we need this?

File: npm/webpack-dev-server/src/plugin.ts [hunk 2]
```diff
@@ -68,7 +64,7 @@ export default class CypressCTOptionsPlugin implements Plugin {
     }
 
     // Webpack 4
-    (compilation as compilation.Compilation).hooks.normalModuleLoader.tap(
+    compilation.hooks.normalModuleLoader.tap(
```
Comment (lmiller1990) [inline npm/webpack-dev-server/src/plugin.ts:67]: since we typed on line 43 we don't need the `as`

File: npm/webpack-dev-server/src/plugin.ts [hunk 3]
```diff
@@ -13,12 +13,12 @@ interface CypressOptions {
   devserverEvents?: EventEmitter
 }
 
-type CypressCTWebpackContext = {
+interface CypressCTWebpackContext extends compilation.Compilation {
```
Comment (lmiller1990) [inline npm/webpack-dev-server/src/plugin.ts:16]: extending interfaces seems more idiomatic.

also:

- https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections
- https://stackoverflow.com/questions/37233735/typescript-interfaces-vs-types

**Review by JessicaSachs — COMMENTED** [review]
A little nervous with these changes because I think the Compilation types are different between webpack 4 and 5,

File: npm/webpack-dev-server/src/plugin.ts (hunk 1 above)
Comment (JessicaSachs) [reply npm/webpack-dev-server/src/plugin.ts:32]: I have no idea. It's from the copy/paste Bart imported

**Pull request merged**